src/jit/codegenarm64.cpp

```diff
@@ -2420,23 +2420,27 @@ CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
             // Floating point divide never raises an exception
             genCodeForBinary(treeNode);     
         }
-        else  // integer divide operation
+        else  // an integer divide operation
         {
             GenTreePtr divisorOp = treeNode->gtGetOp2();
+            emitAttr   size      = EA_ATTR(genTypeSize(genActualType(treeNode->TypeGet())));
 
             // TODO-ARM64-CQ: Optimize a divide by power of 2 as we do for AMD64
             
             if (divisorOp->IsZero())
             {
-                emitJumpKind jmpEqual = genJumpKindForOper(GT_EQ, CK_SIGNED);
-                genJumpToThrowHlpBlk(jmpEqual, SCK_DIV_BY_ZERO);
-                // We don't need to generate the sdiv/udiv instruction
+                // We unconditionally throw a divide by zero exception
+                genJumpToThrowHlpBlk(EJ_jmp, SCK_DIV_BY_ZERO);
+
+                // We still need to produce a register result 
+                emit->emitIns_R_R(INS_mov, size, targetReg, REG_ZR);
```

Comment:
On second thought, are we sure that codegen needs to handle divisor being a constant integer zero?  Isn't morph ensuring div-by-zero gets morphed into a 'Throw'  so that the basic block will be marked to end with a throw and that might lead to dead code elimination?

If front-end ensures the invariant that backend never sees div-by-zero constant, we just need to add an assert here.